### DISCUSSION TITLE
Gödel, Escher, Bach (Jan-May 2023)

### DISCUSSION THREAD

**Discussion by CuriousCurmudgeon** [Virtual-Coffee/virtualcoffee.io#789] [VC Book Club]
Every Monday, I will try to post discussion questions. The pace is negotiable, but I'm looking to take my time on this one so thoughts can stew in the back of my mind. The goal is one chapter plus dialog per week.

### Part 1: GEB
| Dates | Dialog | Chapter |
| ------ | ------- | --------- |
| Dec ??? - Jan 1 | Three-Part Invention | Preface, Intro, Chapter 1: The MU-puzzle |
| Jan 2 - Jan 8 | Two-Part Invention | Chapter 2: Meaning and From in Mathematics |
| Jan 9 - Jan 15 | Sonata for Unaccompanied Achilles | Chapter 3: Figure and Ground |
| Jan 16 - Jan 22 | Contracrostipunctus | Chapter 4: Consistency, Completeness, and Geometry |
| Jan 23 - Jan 29 | Little Harmonic Labyrinth | Chapter 5: Recursive Structures and Processes |
| Jan 30 - Feb 5 | Canon by Intervallic Augmentation | Chapter 6: The Location of Meaning |
| Feb 6 - Feb 12 | Chromatic Fantasy, and Feud | Chapter 7: The Propositional Calculus |
| Feb 13 - Feb 19 | Crab Canon | Chapter 8: Typographical Number Theory |
| Feb 20 - Feb 26 | A Mu Offering | Chapter 9: Mumon and Gödel |

### Part 2: EGB
| Dates | Dialog | Chapter | Additional |
| ------ | ------- | --------- | ----------- |
| Feb. 27 - Mar 5 | Prelude | Chapter 10: Levels of Description, and Computer Systems |
| Mar 6 - Mar 12 | ... Ant Fugue | Chapter 11: Brains and Thoughts |
| Mar 13 - Mar 19 | English French German Suite | Chapter 12: Minds and Thoughts |
| Mar 20 - Mar 26 | Aria with Diverse Variations | Chapter 13: BlooP and FlooP and GlooP |
| Mar 27 - Apr 2 | Air on G's String | Chapter 14: On Formally Undecidable Propositions of TNT and Related Systems |
| Apr 3 - Apr 9 | Birthday Cantatatata... | Chapter 15: Jumping out of the System |
| Apr 10 - Apr 16 | Edifying Thoughts of a Tobacco Smoker | Chapter 16: Self-Ref and Self-Rep |
| Apr 17 - Apr 23 | The Magnificrab, Indeed | Chapter 17: Church, Turing, Tarski, and Others |
| Apr 24 - Apr 30 | SHRDLU, Toy of Man's Designing | Chapter 18: Artificial Intelligence: Retrospects | 
| May 1 - May 7 | Contrafactus | Chapter 19: Artificial Intelligence: Prospects |
| May 8 - May 14 | Sloth Canon | Chapter 20: Strange Loops, Or Tangled Hierarchies | Six-Part Ricercar |

**Comment by CuriousCurmudgeon** [comment]
### Dec ??? - Jan 1: Preface, Intro, Three-Part Invention, Chapter 1: The MU-puzzle
- "[Systems thinking](https://en.wikipedia.org/wiki/Systems_thinking)"  is a way of making sense of the complexity of the world by looking at it in terms of wholes and relationships rather than by splitting it down into its parts. How does this relate to the idea of "strange loops"?
- Does Gödel's Theorem feel strange to you? Or has it been absorbed into our culture to the point where it feels natural?
- In the intro, the author lists out essential capabilities for intelligence. How does today's state-of-the-art AI compare?
- Did you try the MU-puzzle? If so, what progress did you make? What patterns did you discover?
- The author says "that is *possible* for a machine to act unobservant; it is impossible for a human to act unobservant". How does that observation sit with you?

**Reply by cmcrawford2** [reply]
Hi! I found my husband's copy of the book and started reading it. Cannot answer any of the questions yet.

**Reply by cmcrawford2** [reply]
OK I have read up some, here are my answers. I never understood Gödel's theorem (nor did I try). Hopefully I will by the end of the book.
I don't think a computer can come up with a novel idea. That's the last thing on his list. A computer can do the other things.
I didn't try the MU puzzle other than briefly. It seemed impossible to get to MU.
I am going on to chapter 2 and realizing I didn't get everything in chapter 1 so I will go back. I looked up ADAC, sorry to say.

**Reply by cmcrawford2** [reply]
"It is impossible for a human to act unobservant." I agree. Everything we look at has a meaning that we have given it. We can't look at anything without applying a meaning to it. That meaning is the "model" that we create in order to deal with reality. I don't think it's possible for a machine to act unobservant, though I might not understand exactly what he means by that.

**Reply by tomcudd** [reply]
I've often thought that systems which expand to a holistic view have an elegance to them when they try to "self-contain" the importance pieces of those systems. I look at eco systems as an interesting representations of strange loops in the natural world. However, I do see that many attempts to obfuscate, encompass, or explain in metaphors eventually break down. So I'm trying to reconcile that. I think Gödel's work is felt or internalized now. Humans generally think that human made systems have edge cases that break down and we seem to think that more of systems than not.

**Reply by CuriousCurmudgeon** [reply]
> I think Gödel's work is felt or internalized now. Humans generally think that human made systems have edge cases that break down and we seem to think that more of systems than not.

This might be a stretch, but I wonder how this relates to declining trust in human made institutions. I don't know how it was viewed at the time, but, in retrospect, Russell and Gödel's time was defined by a lot of institution building and trust in those institutions. That is rapidly crumbing in our time. The evidence that human made systems break down in certain conditions is all around us now.

**Reply by tomcudd** [reply]
> This might be a stretch, but I wonder how this relates to declining trust in human made institutions.

I don't think it's a stretch at all. I can imagine the shock of the systems that broke down in the past, whereas something as complex and critical as supply chains breaking down now is almost ho hum and registers as an annoyance to the consumer of products

**Comment by CuriousCurmudgeon** [comment]
Jan 2 - Jan 8: Two-Part Invention, Chapter 2: Meaning and Form in Mathematics

- How did you view the moving of goalposts by the Tortoise in the dialog with Achilles?
- How strong was your desire to over apply the isomorphism between the pq-system and addition? Did you want to assign even more meaning to the pattern match your brain had already picked up on?

**Reply by tomcudd** [reply]
To me, the moving of the goalposts just felt like an opportunity to try and educate, but I think most reasonable average modern humans, would understand the concepts and give up fast. It's interesting to sit in something that's infinite, asymptotic, or otherwise not ending to really start to understand.

In tech, I feel like attempting to apply isomorphism to new situations is a common effort. I'm always trying to apply meaning to patterns. I started going all in on trying to apply what I understand about addition within mathematical systems to the pq- system.

**Comment by CuriousCurmudgeon** [comment]
Jan 9 - Jan 15: Sonata for Unaccompanied Achilles, Chapter 3: Figure and Ground

Going forward, I will unashamedly be borrowing discussion questions from the [GEB wiki](https://godel-escher-bach.fandom.com/wiki/G%C3%B6del,_Escher,_Bach_Wiki) that I found last night. Feel free to discuss whatever aspects of the reading interested you and how they relate to anything else in your life.

1. What was the tortoise doing that caused Achilles to twist his neck?
2. What was the tortoise's ingenious but degenerate solution to the HE puzzle?
3. What was the tortoise's second solution to the HE puzzle?
4. What is the solution to the ADAC puzzle?
5. People tend to believe intuitively, when they learn about prime numbers, that 1 is prime. According to the definitions used in number theory, however, 1 is neither prime nor composite. What ensures that 1 is not prime in the DND-system -- that is, that `P-` is not a theorem?
6. What would happen to the DND-system if `P-` were an axiom?
7.  In the bottom-up decision procedure described on p. 48, why is it important that the axioms are ordered with the simplest (shortest) axiom first? What could go wrong if the axioms were put into the bucket in a different order?
8.  Why is the FIGURE-FIGURE Figure shaped like that? If you think it's just an arbitrary shape that tiles the plane, reread p. 67.
9.  Solve the puzzle on p. 73. What number comes next in the sequence 1, 3, 7, 12, 18, 26, 35, 45, 56, 69, ...? How is this like the FIGURE-FIGURE Figure?
10. Is the MIU-system recursive? Can you make a set of rules that characterize its negative space?

**Reply by tomcudd** [reply]
I quite enjoyed the negative space telephone conversation. I'll be honest, a huge part of problem solving in my day to day work is built on seeing what isn't there and trying to fill it in to figure out why something isn't working. Trying to formalize it in a system, bringing some kind of ordered description is part of the work. Starting in an ordered manner (lowest level?) is critical to not "miss" the thing that's wrong or what you're searching for.

**Reply by tomcudd** [reply]
I think the answer is 73 for the sequence.

**Comment by CuriousCurmudgeon** [comment]
### Jan 16 - Jan 22: Contracrostipunctus, Chapter 4: Consistency, Completeness, and Geometry

1. What's the hidden message in this dialogue? What's the second hidden message?
2. What kind of record do you think would break Record Player Omega?
3. Why does DRH keep apologizing about his use of the word "isomorphism"?
4. Give an example of when a word can have multiple meanings. When you say something, how do you communicate the intended meaning?
5. Do you consider mathematics to be the same in all possible worlds? Consider carefully what you consider to be mathematics.
6. Do you consider logic to be the same in all possible worlds? What, then, constitutes a possible world?
7. Answer the question on p. 102 about the record and the record player. How can they both pass their respective tests?

#### Additional Resources
These wiki links answer some of the questions above, but are still useful to drive discussion and understanding.
- https://godel-escher-bach.fandom.com/wiki/Contracrostipunctus
- https://godel-escher-bach.fandom.com/wiki/Chapter_4

**Reply by tomcudd** [reply]
The intended meaning of words that can have multiple meanings depend heavily on: context clues, non verbal cues, or sometimes explicit definition. Examples are sanction, cleave, buckle. I often think of sarcasm in our daily lives and how it relies heavily on these other clues.

**Comment by CuriousCurmudgeon** [comment]
### Jan 23 - Jan 29: Little Harmonic Labyrinth, Chapter 5: Recursive Structures and Processes

1. Why does each Meta-Genie perform its task "twice as quickly" as the Genie before it?
2. What are some other contexts that involve recursion?
3. How does human intelligence makes use of recursion, according to Hofstadter?

#### Additional Resources
These wiki links answer some of the questions above, but are still useful to drive discussion and understanding.
- https://godel-escher-bach.fandom.com/wiki/Little_Harmonic_Labyrinth
- https://godel-escher-bach.fandom.com/wiki/Chapter_5

**Reply by tomcudd** [reply]
It's my understanding that intelligence makes use of complicated recursion that increases in complexity, thus breaking out of a defined pattern. Think about driving. Essentially, it's minor recursive adjustments based on input to the driver. Consider a daily commute route and what happens when construction closes a bridge. Our recursive driving of that route settles into a new "steady state" as we learn a new route. Only based on new input will our recursive model for "getting to work" change. At least that's sort of how I envision what's being said here.

**Comment by CuriousCurmudgeon** [comment]
### Jan 30 - Feb 5: Canon By Intervallic Augmentation, Chapter 6: The Location of Meaning

1. Is meaning the interaction of a mind (or mechanism) and a message, or is it inherent in the message?
2. What is being "pulled out" of DNA?
3. To what extent do the mathematical equations in physics "mean" the behavior of the universe?

#### Additional Resources
These wiki links answer some of the questions above, but are still useful to drive discussion and understanding.
- https://godel-escher-bach.fandom.com/wiki/Canon_by_Intervallic_Augmentation
- https://godel-escher-bach.fandom.com/wiki/Chapter_6

**Reply by tomcudd** [reply]
Ultimately, the physical forms and characteristics are what is being pulled from the information of DNA. It's within DNA that instructions are coded.

**Comment by CuriousCurmudgeon** [comment]
### Feb 6 - Feb 12: Chromatic Fantasy, and Feud, Chapter 7: The Propositional Calculus

1. Why does the author use fragments of Zen koans as propositions?
2. What are the different kinds of ways that people deal with contradictions in their own thoughts?

#### Additional Resources
These wiki links answer some of the questions above, but are still useful to drive discussion and understanding.
- https://godel-escher-bach.fandom.com/wiki/Chromatic_Fantasy,_And_Feud
- https://godel-escher-bach.fandom.com/wiki/Chapter_7

**Comment by CuriousCurmudgeon** [comment]
### Feb 13 - Feb 19: Crab Canon, Chapter 8: Typographical Number Theory

1. Even with careful reading and note taking, this chapter broke my brain at several times. Don't feel discouraged if you felt the same.
2. Did you have any luck with the exercises to translate from TNT to English and vice-versa?
3. The author once again talks about M-mode vs. I-mode when comparing applying TNT vs. reasoning about TNT. Chat GPT is very hot in the news these days. Where does it fit into the M vs. I-mode structure?

#### Additional Resources
These wiki links answer some of the questions above, but are still useful to drive discussion and understanding.
- https://godel-escher-bach.fandom.com/wiki/Crab_Canon
- https://godel-escher-bach.fandom.com/wiki/Chapter_8

**Comment by CuriousCurmudgeon** [comment]
### Feb 20 - Feb 26: A Mu Offering, Chapter 9: Mumon and Gödel

1. Zen koans are deliberately non-sensical. What is your immediate reaction when reading one? How does your brain handle the discord?
2. What comparisons did you draw to string folding in the dialog? Does it feel like an isomorphism of those topics?
3. How does Zen relate to the incompleteness theorem?

#### Additional Resources
These wiki links answer some of the questions above, but are still useful to drive discussion and understanding.
- https://godel-escher-bach.fandom.com/wiki/A_Mu_Offering
- https://godel-escher-bach.fandom.com/wiki/Chapter_9

**Comment by CuriousCurmudgeon** [comment]
### Feb 27 - Mar 5: Prelude, Chapter 10: Levels of Description, and Computer Systems

1. How does modern chess AI work today vs. when GEB was published in 1979?
2. Where does Chat GPT fall in the classification of "rubbery languages"? Are systems like this an entirely new category that did not exist at the time?
3. Developers usually work as part of a team. Would you consider these teams to be "nearly decomposable systems"?

#### Additional Resources
These wiki links answer some of the questions above, but are still useful to drive discussion and understanding.
- https://godel-escher-bach.fandom.com/wiki/Prelude
- https://godel-escher-bach.fandom.com/wiki/Chapter_10

**Comment by CuriousCurmudgeon** [comment]
### Mar 6 - Mar 12: Ant Fugue, Chapter 11: Brains and Thoughts

1. Is a state machine a way to give meaning to what can otherwise appear to be a chaotic mess of signals and symbols.?
2. GEB was published in 1979. The author is clear that very little was actually known about how our brains work at the time. How have the ideas explored in this chapter held up?
3. How does a symbol sending out messages relate to programming?
4. Has neuroscience reached the point where we can describe symbols in our brain at a higher level, completely abstracting away the neurons?

**Comment by CuriousCurmudgeon** [comment]
### Mar 13 - Mar 19: English French German Suite, Chapter 12: Minds and Thoughts

1. There are many organizational systems, such as Zettelkasten, that attempt to link together your notes in a useful way. How similar is this to building a semantic network from your thoughts? How dissimilar?
2. Automated language translation has come along way, but is still far from perfect. How do you think early optimism around translation has aged? What has proven harder than expected?
3. How do you think Turing's view of the "critical" computing size for intelligent machines compares to the amount of compute we have available today? Would he be surprised that general AI is still not possible?

**Comment by CuriousCurmudgeon** [comment]
### Mar 20 - Mar 26: Aria with Diverse Variations, Chapter 13: BlooP and FlooP and GlooP

1. Are you familiar with the halting problem? If so, how did that impact your reading this week?
2. Why is the Tortoise property harder to reliably compute than the Goldbach conjecture?
3. What patterns do you see when calculating if a number is "wonderous"?
4. I found it difficult to understand the diagonal method proof for Blue Programs, but much easier to understand Cantor's original diagonal argument. How did the two proofs strike you?

**Comment by CuriousCurmudgeon** [comment]
### Mar 27 - Apr 2: Air on G's String, Chapter 14: On Formally Undecidable Propositions of TNT and Related Systems

1. Does a quine about quines qualify as a strange loop?
2. How does the incompleteness of TNT relate to the incompleteness of absolute geometry?

**Comment by CuriousCurmudgeon** [comment]
### Apr 3 - Apr 9: Birthday Cantatatata, Chapter 15: Jumping out of the System

1. The author notes that "once a system is well-defined, or 'boxed', it becomes vulnerable". He's referring to number theory, but what other types of systems could this apply to?
2. Where do you fall in the debate over the mechanical mind? Is the human mind able to perform computations that a computer will never be able to do?

**Comment by CuriousCurmudgeon** [comment]
### Apr 10 - Apr 16: Edifying Thoughts of a Tobacco Smoker, Chapter 16: Self-Ref and Self-Rep

1. GEB was published in 1979. How do you think advancements in genetic research affect this chapter? Does the central tenet the different levels of cellular biology are woven together into a self-referential whole still hold?
2. How does the isomorphism between molecular binding/recognition and theoremhood feel to you?

**Comment by CuriousCurmudgeon** [comment]
### Apr 17 - Apr 23: The Magnificrab, Indeed, Chapter 17: Church, Turing, Tarski, and Others

1. Is it inevitable that we do not understand the inner workings of sufficiently powerful AI models? How does this relate to understanding the nature of our own intelligence?
2. What has the current generation of LLMs (large language models) revealed about what we consider to be "real" intelligence?

**Reply by tomcudd** [reply]
I am reminded of the first time that I learned about machine learning via neural networks. It still baffles me that we can look at the weights in nodes and still not be sure how an outcome is derived that correctly solves a problem or identifies something via a nueral network. It really does show humans that we may have trouble identifying what real intelligence is, let alone artificial.

**Comment by CuriousCurmudgeon** [comment]
### Apr 24 - Apr 30: SHRDLU, Toy of Man's Designing, Chapter 18: Artificial Intelligence: Retrospects

1. Early computer inventors experienced an almost mystical sense of being in the presence of another kind of "thinking being". Do you think modern AI researchers feel the same about cutting edge AI today?
2. How long will it take before large language models begin to feel blasé?
3. The chapter makes a couple of mentions of tasks that researchers thought AI would easily be able to solve. Do you think they solicited input from experts in those fields? Do you think AI researchers risk falling into the same trap today?
4. The author states that "it is probably safe to say that writing a program which can fully handle the top five words of English - 'the', 'of', 'and', 'a', and 'to' - would be equivalent to solving the entire problem of AI and hence tantamount to knowing what intelligence and consciousness are." How has this take aged?

**Comment by CuriousCurmudgeon** [comment]
### May 1 - May 7: Contrafactus, Chapter 19: Artificial Intelligence: Prospects

1. The ability to lie is an important developmental milestone in children. It shows they can see the world as it is and invent an alternate reality. Current LLM models like Chat GPT are also notorious for lying. In what way is this different than childhood development?
2. The chapter ends with ten questions and speculations. How have these takes aged?

**Comment by CuriousCurmudgeon** [comment]
### May 8 - May 14: Sloth Canon, Chapter 20: Strange Loops, Or Tangled Hierarchies, Six-Part Ricercar

We're at the end! Here are the final discussion questions.

1. Other animals are not intelligent enough to understand their own brains. How presumptive is it of us to assume that we are intelligent enough to understand our own?
2. "A Tangled-Hierarchy occurs when what you presume are clean hierarchical levels take you by surprise and fold back in a hierarchy violating way." What was your favorite example of a Tangled-Hierarchy from the book?
